interfaces/hotplug/deviceinfo.go

```diff
@@ -0,0 +1,113 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+type HotplugDeviceInfo struct {
+	object       string
+	Data         map[string]string
+	idVendor     string
```

Comment:
I'm somewhat -0.5 on keeping those explicit. Instead we could have methods that fish them out of `Data`

Reply:
Note, these attributes are not present in Data, they were read lazily on first access from the filesystem. However, removed all the idVendor/idProduct/serial aspects for now per our discussion.